core/src/main/scala/io/finch/request/package.scala

```diff
@@ -192,7 +218,76 @@ package object request {
      */
     def shouldNot(rule: ValidationRule[A]): RequestReader[A] = shouldNot(rule.description)(rule.apply)
   }
-
+  
+  /**
+   * Implicit conversion that allows to call ''as[A]'' on any ''RequestReader[String]''
+   * to perform a type conversion based on an implicit ''DecodeRequest[A]'' which must
+   * be in scope.
+   * 
+   * The resulting reader will fail when type conversion fails.
+   */
+  implicit class StringReaderOps(val reader: RequestReader[String]) {
+
+    /* IMPLEMENTATION NOTE: The three implicit classes for string based readers should
+     * all extend AnyVal to avoid instance creation for each invocation of the extension
+     * method. However, this let's us run into a compiler bug when we compile for Scala 2.10:
+     * https://issues.scala-lang.org/browse/SI-8018. The bug is caused by the combination of
+     * three things: 1) an implicit class, 2) extending AnyVal, 3) wrapping a class with type
+     * parameters. 2) is the only thing we can remove here, otherwise we'd need to move the
+     * body of the methods somewhere else. Once we drop support for Scala 2.10, these classes
+     * can safely extends AnyVal. 
+     */
```

Comment:
Great explanation. We probably want to create an issue to track this `AnyVal` is added back to these implicit class when scala 2.10 support is dropped.